jans-auth-server/common/src/test/java/io/jans/as/common/RedirectUriTest.java

```diff
@@ -0,0 +1,342 @@
+        AbstractCryptoProvider cryptoProvider = mock(AbstractCryptoProvider.class);
+        when(cryptoProvider.getPublicKey(anyString(), Mockito.any(), Mockito.any())).thenReturn(getRSAPublicKey());
+        when(cryptoProvider.sign(anyString(), anyString(), anyString(), anyObject())).thenReturn("12345");
+
+        Jwe jwe = mock(Jwe.class);
+        when(jwe.encryptJwe(any(), any())).thenReturn(new Jwe());
+        redirectUri.setCryptoProvider(cryptoProvider);
+        redirectUri.setJwtForEncrypt(jwe);
+        redirectUri.setNestedKeyId("nestedKey123");
+        redirectUri.setNestedSharedSecret("nested_shared_secret");
+
+        String queryResult = redirectUri.getQueryString();
+
+        assertNoEmptyQueryString(queryResult, RESPONSE, 1);
+    }
+
+    @Test()
+    public void getQueryString_withEncriptionAlgorithm128KWAndSignatureAlgorithm_responseEncoded() throws UnsupportedEncodingException, CryptoProviderException, InvalidJweException {
+        RedirectUri redirectUri = getRedirectUriTemplateGetQueryString(ResponseMode.JWT, KeyEncryptionAlgorithm.A128KW, BlockEncryptionAlgorithm.A128GCM, SignatureAlgorithm.HS256);
+
+        AbstractCryptoProvider cryptoProvider = mock(AbstractCryptoProvider.class);
+        Jwe jwe = mock(Jwe.class);
+        when(cryptoProvider.sign(anyString(), anyString(), anyString(), any(SignatureAlgorithm.class))).thenReturn("12345");
+        when(jwe.encryptJwe(any(), any())).thenReturn(new Jwe());
+
+        redirectUri.setCryptoProvider(cryptoProvider);
+        redirectUri.setJwtForEncrypt(jwe);
+        redirectUri.setNestedKeyId("nestedKey123");
+        redirectUri.setNestedSharedSecret("nested_shared_secret");
+
+        String queryResult = redirectUri.getQueryString();
+
+        assertNoEmptyQueryString(queryResult, RESPONSE, 1);
+
+    }
+
+    @Test()
+    public void getQueryString_noEncriptionAlgorithmNoSignatureAlgorithm_responseEncoded() throws UnsupportedEncodingException, CryptoProviderException, InvalidJweException {
+        RedirectUri redirectUri = getRedirectUriTemplateGetQueryString(ResponseMode.JWT, null, null, null);
+
+        AbstractCryptoProvider cryptoProvider = mock(AbstractCryptoProvider.class);
+        when(cryptoProvider.sign(anyString(), anyString(), anyString(), any(SignatureAlgorithm.class))).thenReturn("12345");
+        redirectUri.setCryptoProvider(cryptoProvider);
+
+        String queryResult = redirectUri.getQueryString();
+
+        assertNoEmptyQueryString(queryResult, RESPONSE, 1);
+
+    }
+
+    @Test()
+    public void toString_withResponseModeFormPostJwtAndNoParams_NoEmpty() {
+
+        RedirectUri redirectUri = getRedirectUriTemplateToString();
+        redirectUri.setResponseMode(ResponseMode.FORM_POST_JWT);
+        redirectUri.setCryptoProvider(cryptoProvider);
+
+        assertTrue(redirectUri.toString().length() > 0, "Is Empty");
```

Comment:
toString() is important test. In each `toString()` test we should see equals assertion
```
assertEquals("expected value", redirectUri.toString());
```
Also when I look into that method after your refactoring I am not sure logic of the method is not broken. If you cover it with tests correctly (with equals) we will know it exactly. It is main goal of writing these tests. It is not to get good coverage number but really be sure that logic inside is exactly what we expect it to be.